tool/docs/java/JavaDocsParser.kt

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.typedb.client.tool.doc.java
+
+import com.vaticle.typedb.client.tool.doc.common.*
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+
+fun main(args: Array<String>) {
+    val inputDirectoryName = args[0]
+    val outputDirectoryName = args[1]
+
+    val docsDir = System.getenv("BUILD_WORKSPACE_DIRECTORY")?.let { Paths.get(it).resolve(outputDirectoryName) }
+        ?: Paths.get(outputDirectoryName)
+    if (!docsDir.toFile().exists()) {
+        Files.createDirectory(docsDir)
+    }
+
+    File(inputDirectoryName).walkTopDown().filter {
+        it.toString().contains("/api/") && !it.toString().contains("-use")
+                && !it.toString().contains("-summary") && !it.toString().contains("-tree")
+                && it.toString().endsWith(".html")
```

Comment:
Let's discuss this to see how we can parametrise this, rather than hard-coding it in here. Same with the other parsers